src/imgui/ImGuiMachine.cc

```diff
@@ -150,7 +157,46 @@ void ImGuiMachine::showMenu(MSXMotherBoard* motherBoard)
 
 			loadSetupOpen = setupFileList.menu("Load setup ...");
 
-			saveSetupOpen = im::Menu("Save setup ...", true, [&]{
+			saveSetupOpen = im::Menu("Save current setup as ...", true, [&]{
+				ImGui::TextUnformatted("Save current setup:");
+
+				using enum MSXMotherBoard::SetupDepth;
+
+				auto depthToGuiString = [](const MSXMotherBoard::SetupDepth depth) {
+					if (depth == MACHINE_ONLY                            ) return MACHINE_NODE_NAME;
+					if (depth == WITH_EXTENSIONS                         ) return EXTENSIONS_NODE_NAME;
+					if (depth == WITH_EXTENSIONS_AND_PLUGGABLES          ) return CONNECTORS_NODE_NAME;
+					if (depth == WITH_EXTENSIONS_AND_PLUGGABLES_AND_MEDIA) return MEDIA_NODE_NAME;
+					if (depth == COMPLETE_STATE                          ) return STATE_NODE_NAME;
+					return "";
+				};
+				auto depthToHelpString = [](const MSXMotherBoard::SetupDepth depth) {
+					if (depth == MACHINE_ONLY                            ) return "Only the machine itself, without anything in it.";
+					if (depth == WITH_EXTENSIONS                         ) return "The machine with all plugged in extensions.";
+					if (depth == WITH_EXTENSIONS_AND_PLUGGABLES          ) return "The machine, with all plugged in extensions and all things that are plugged into the connectors.";
+					if (depth == WITH_EXTENSIONS_AND_PLUGGABLES_AND_MEDIA) return "The machine, with all plugged in extensions and all things that are plugged into the connectors and also all inserted media.";
+					if (depth == COMPLETE_STATE                          ) return "The full state of the machine, with everything that's in it at the current time.";
+					return "";
+				};
+
+				im::Combo("Select level",
+						tmpStrCat(saveSetupDepth == MACHINE_ONLY ? "" : "+ ", depthToGuiString(saveSetupDepth)).c_str(), [&]{
+					for (const auto& d : { MACHINE_ONLY, WITH_EXTENSIONS,
+							       WITH_EXTENSIONS_AND_PLUGGABLES,
+							       WITH_EXTENSIONS_AND_PLUGGABLES_AND_MEDIA,
+							       COMPLETE_STATE }) {
+						if (d != MACHINE_ONLY) {
+							ImGui::Indent(ImGui::CalcTextSize("m").x);
```

Comment:
This is unexpected to me, but apparently it works ;)

I had the impression that `ImGui::Indent()` should always be paired with `ImGui::Unindent()` (and you can enforce this via our `im::Indent()` wrapper. So my expectation was something like:
```
im::Indent(float(depth) * some_value, []{
 ...
});
```
But when I check the implementation of these Dear ImGui functions, my required-pairing-expectation seems invalid. And then your implementation is better.

Maybe one detail: move the calculation of `ImGui::CalctextSize()` outside the loop:
```
auto indentSize = ...
```
Text size calculation is not super expensive, but certainly also not trivial (taking unicode encoding into account, proportional fonts, (in the future) partially loaded fonts, ...).

Reply:
Good point. Done.